### PR TITLE
Data Refresh and Display Issues

### PR DESCRIPTION
**Pull request by happycodemonkey**
## Overview

Fixes multiple small issues with extensions listing including submit not refreshing data display properly and small styling issues. 

## Related

[WP-739](https://tacc-main.atlassian.net/browse/WP-739)

## Changes

- edit modal should properly display data and new data should be refetched on submit
- view modal should display data properly
- both modals should have styling more consistent with production

## Testing

1. go to http://localhost:8000/administration/list-extensions/
2. open edit modal and verify record has an Applicable Data Period (a few records do not, this is simply bad data, please disregard and select another that has this field populated)
3. Edit Applicable Data Period and verify it successfully submits, and the Applicable Data Period display field under the form both updates and displays properly
4. Edit any other field and verify the same
5. Close the modal and make sure the record was properly update in the listing
6. Verify both Edit and View modals seem consistent with other modals
7. Verify other extensions are displaying data properly even after first record has been viewed/edited

## UI

…

### CODE REVIEW

File: apcd-cms/src/client/src/components/Extensions/EditExtensionModal/EditExtensionModal.tsx [hunk 1]
```diff
@@ -100,22 +100,28 @@ const EditExtensionModal: React.FC<EditExtensionModalProps> = ({
...
       const response = await fetchUtil({
         url,
         method: 'PUT',
-        body: values,
+        body: api_values,
       });
 
       if (onEditSuccess && response) {
+          console.log('success, setting field');
+          console.log(convertApiValueToPeriodLabel(values.applicable_data_period));
```
Comment (chandra-tacc) [inline apcd-cms/src/client/src/components/Extensions/EditExtensionModal/EditExtensionModal.tsx:119]:
```suggestion
```

File: apcd-cms/src/client/src/utils/dateUtil.ts [hunk 2]
```diff
@@ -62,3 +62,43 @@ export const convertPeriodLabelToApiValue = (period: string): string | null => {
...
+  if (!match) {
+    console.log(`Invalid period format: ${period}`);
+    return period;
+  }
+
+  const [_, year, month] = match;
+  const stringMonth = monthMap[month];
+
+  if (!stringMonth) {
+    console.log(`Invalid month: ${month} in ${period}`);
```
Comment (chandra-tacc) [inline apcd-cms/src/client/src/utils/dateUtil.ts:99]:
```suggestion
```

File: apcd-cms/src/client/src/utils/dateUtil.ts [hunk 3]
```diff
@@ -62,3 +62,43 @@ export const convertPeriodLabelToApiValue = (period: string): string | null => {
...
+    '09': 'Sep',
+    '10': 'Oct',
+    '11': 'Nov',
+    '12': 'Dec',
+  };
+
+  const match = period.match(/^(\d{4})-(0[1-9]|1[0-2])$/);
+  //const match = period.match(/^([A-Za-z]{3})\.?\s(\d{4})$/);
+  if (!match) {
+    console.log(`Invalid period format: ${period}`);
```
Comment (chandra-tacc) [inline apcd-cms/src/client/src/utils/dateUtil.ts:91]:
```suggestion
```

**Review by chandra-tacc — APPROVED** [review]
LGTM, minor comments.

**Review by sophia-massie — APPROVED** [review]
This looks great! The only thing not refreshing is the helper text once you submit. Not a blocker. 

I was trying to crack the data refresh for extension modal to try and help, but only got through some styling. If you want to merge these changes in a PR I created from that debug session, feel free. 

I branched off you PR so that this modal looks like the exception modal [here.](https://github.com/TACC/Core-CMS-Custom/pull/371) You can merge into your PR, and then merge into the feature branch, or skip it. Totally up to you. There are no merge conflicts to sort through.

**Comment by sophia-massie** [comment]
LGTM with recent changes 👍

**Pull request merged**